detection_rules/rule.py

```diff
@@ -152,18 +153,58 @@ def validate(self, as_rule=False, versioned=False, query=True):
 
         schema_cls.validate(contents, role=self.type)
 
-        if query and self.query and self.contents['language'] == 'kuery':
+        if query and self.query is not None:
             ecs_versions = self.metadata.get('ecs_version')
             indexes = self.contents.get("index", [])
-            self._validate_kql(ecs_versions, indexes, self.query, self.name)
+
+            if self.contents['language'] == 'kuery':
+                self._validate_kql(ecs_versions, indexes, self.query, self.name)
+
+            if self.contents['language'] == 'eql':
+                self._validate_eql(ecs_versions, indexes, self.query, self.name)
+
+    @staticmethod
+    @cached
+    def _validate_eql(ecs_versions, indexes, query, name):
+        # validate against all specified schemas or the latest if none specified
+        parsed = eql.parse_query(query)
```

Comment:
can you change the `parsed_kql` property to 

```python
    def parsed_query(self):
        language = self.contents.get('language')
        if self.query and language in ('kuery', 'eql'):
            return kql.parse(self.query) if language == 'kuery' else eql.parse_query(self.query)
```

Then you can use that here (and more consistently as needed)

Reply:
i can't use it here because this is a static method and needs to be for caching to work.

but i did update the method regardless, even though it's never used